samples/bluetooth/llpm/src/main.c

```diff
@@ -0,0 +1,412 @@
+		.interval = 0x0010,
+		.window = 0x0010,
+	};
+
+	struct bt_scan_init_param scan_init = {
+		.connect_if_match = true,
+		.scan_param = &scan_param,
+		.conn_param = conn_param
+	};
+
+	bt_scan_init(&scan_init);
+	bt_scan_cb_register(&scan_cb);
+
+	err = bt_scan_filter_add(BT_SCAN_FILTER_TYPE_UUID, BT_UUID_LATENCY);
+	if (err) {
+		printk("Scanning filters cannot be set\n");
+		return;
+	}
+
+	err = bt_scan_filter_enable(BT_SCAN_UUID_FILTER, false);
+	if (err) {
+		printk("Filters cannot be turned on\n");
+	}
+}
+
+static void discovery_complete(struct bt_gatt_dm *dm, void *context)
+{
+	struct bt_gatt_latency *latency = context;
+
+	printk("Service discovery completed\n");
+
+	bt_gatt_dm_data_print(dm);
+	bt_gatt_latency_handles_assign(dm, latency);
+	bt_gatt_dm_data_release(dm);
+
+	/* Start testing when the GATT service is discovered */
+	test_ready = true;
+}
+
+static void discovery_service_not_found(struct bt_conn *conn, void *context)
+{
+	printk("Service not found\n");
+}
+
+static void discovery_error(struct bt_conn *conn, int err, void *context)
+{
+	printk("Error while discovering GATT database: (%d)\n", err);
+}
+
+struct bt_gatt_dm_cb discovery_cb = {
+	.completed         = discovery_complete,
+	.service_not_found = discovery_service_not_found,
+	.error_found       = discovery_error,
+};
+
+static void advertise_and_scan(void)
+{
+	int err;
+
+	err = bt_le_adv_start(BT_LE_ADV_CONN, ad, ARRAY_SIZE(ad),
+			      sd, ARRAY_SIZE(sd));
+	if (err) {
+		printk("Advertising failed to start (err %d)\n", err);
+		return;
+	}
+
+	printk("Advertising successfully started\n");
+
+	err = bt_scan_start(BT_SCAN_TYPE_SCAN_PASSIVE);
+	if (err) {
+		printk("Starting scanning failed (err %d)\n", err);
+	}
+	printk("Scanning successfully started\n");
+}
+
+static void connected(struct bt_conn *conn, u8_t err)
+{
+	if (err) {
+		printk("Connection failed (err %u)\n", err);
+		return;
+	}
+
+	default_conn = bt_conn_ref(conn);
+	err = bt_conn_get_info(default_conn, &conn_info);
+	if (err) {
+		printk("Error %d while getting bt conn info\n", err);
+	}
+
+	printk("Connected as %s\n",
+	       conn_info.role == BT_CONN_ROLE_MASTER ? "master" : "slave");
+	printk("Conn. interval is %u units (1.25 ms/unit)\n",
+	       conn_info.le.interval);
+
+	/* make sure we're not scanning or advertising */
+	bt_le_adv_stop();
+	bt_scan_stop();
+
+	err = bt_gatt_dm_start(default_conn, BT_UUID_LATENCY, &discovery_cb,
+			       &gatt_latency);
+	if (err) {
+		printk("Discover failed (err %d)\n", err);
+	}
+}
+
+static void disconnected(struct bt_conn *conn, u8_t reason)
+{
+	printk("Disconnected (reason %u)\n", reason);
+
+	test_ready = false;
+
+	if (default_conn) {
+		bt_conn_unref(default_conn);
+		default_conn = NULL;
+	}
+	advertise_and_scan();
+}
+
+static void le_param_updated(struct bt_conn *conn, u16_t interval,
+			     u16_t latency, u16_t timeout)
+{
+	if (interval == INTERVAL_LLPM) {
+		printk("Connection interval updated: LLPM (1 ms)\n");
```

Comment:
What about a situation, when different interval than 1 ms was set. We do not need to inform about it ?

Reply:
In this sample, the le_param_update is only used to enable the llpm interval which is fixed to 1 ms. Moreover, this LLPM interval is a predefined value and it is far from the definition in BLE spec

Reply:
So is not possible to get for some reason here at other value of interval which is invalid for this sample ?

Reply:
Per discussion, this sample sets INTERVAL_LLPM to both interval_min and interval_max, it won't get a callback if the interval is unnegotiable with its peer.